sound/soc/sof/topology.c

```diff
@@ -1303,6 +1351,27 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 		return ret;
 	}
 
+	/* if there is only one source/sink pin, there is no need to parse source/sink
+	 * pin binding from topology.
+	 */
+	if (swidget->num_source_pins > 1) {
+		ret = sof_parse_pin_binding(swidget, priv, true);
+		if (ret < 0) {
+			dev_err(scomp->dev, "failed to parse component pin tokens for %s\n",
```

Comment:
add source in the error message here and sink in the one below to differentiate them maybe?